lib/client.js

```diff
@@ -75,8 +82,22 @@ function messageHandler(client, event) {
       pendingPromise.resolve(data.result);
     }
   } else if (ZAF_EVENT.test(data.key)) {
-    var key = data.key.replace(ZAF_EVENT, '');
-    if (client) {
+    var key = data.key.replace(ZAF_EVENT, ''),
+        msg = { appGuid: client._appGuid };
+    if (data.reply) {
```

Comment:
Ha, I originally used `needsReply` then decided `reply` was cleaner because simpler.  Can change it back